lib/resque.rb

```diff
@@ -255,6 +295,30 @@ def after_pause=(block)
     register_hook(:after_pause, block)
   end
 
+  def before_reserve(&block)
+    block ? register_hook(:before_reserve, block) : hooks(:before_reserve)
+  end
+
+  def before_reserve=(block)
+    register_hook(:before_reserve, block)
+  end
+
+  def after_perform(&block)
+    block ? register_hook(:after_perform, block) : hooks(:after_perform)
+  end
+
+  def after_perform=(block)
+    register_hook(:after_perform, block)
+  end
+
+  def before_push(&block)
+    block ? register_hook(:before_push, block) : hooks(:before_push)
+  end
+
+  def before_push=(block)
+    register_hook(:before_push, block)
+  end
```

Comment:
FEATURE: `before_push`

Blocks passed to this callback will be passed a `Resque::Job` object representing a job about to be put into Redis.

The monolith uses this to attach instrumentation metadata to jobs.

This could be PRed into master.